### ISSUE TITLE
Error dialog fails to pop up when posting a blog entry fails

### ISSUE THREAD

**Issue by joemclain** [RickStrahl/MarkdownMonster#824]
Error dialog fails to pop up when posting a blog entry fails

In:

WeblogAddin.cs
~ line 260
``` c#
	if (!result)
	{
		mmApp.Log($"Error sending post to Weblog at {weblogInfo.ApiUrl}: " + client.ErrorMessage);
		MessageBox.Show(WeblogForm, "Error sending post to Weblog: " + client.ErrorMessage,
			mmApp.ApplicationName,
			MessageBoxButton.OK,
			MessageBoxImage.Exclamation);
			return false;
		}
```
The MessageBox WeblogForm parameter is null. The MessageBox call throws an exception which is caught in WebLogForm.xaml.cs where a status message error message is posted.

The WebglogForm instance should be set in WeblogAddin.cs ~ line 109. The containing method OnExecute() is never called.

Or so it goes on my machine, MM 2.0.3, running under debug.

-joe mclain-

**Comment by joemclain** [comment]
On further checking I find that OnExecute() is called when Weblog is selected from the Addins menu, but not when any of the Weblog menu items are selected.

**Comment by RickStrahl** [comment]
This was fixed some time ago. IT doesn't show error info unfortunately because the XmlRpc library unfortunately doesn't surface the full error information 😢

**Comment by joemclain** [comment]
Um ... As of the source I pulled down an hour ago (6:30 pm PST, 3 Aug 21)

- OnExecute() is ONLY called when Weblog is selected from the Addins menu, it is never called if you try to post an entry without having selected weblog from the Addin menu.
- If the post fails the code attempts to display a MessageBox which fails because WeblogForm is null (because OnExecute() was not called.)

I dropped a few breakpoints and watched this happen when I attempted to post to a blog that I knew would fail.

If this doesn't make sense tell me and I'll try to phrase this better.

-joe mclain-
